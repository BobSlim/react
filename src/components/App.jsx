import { Icon } from "@iconify/react"
import { Outlet, Link } from "react-router-dom"
import { setItem, removeItem } from "./cart"
import { useState, useEffect } from "react"
import { getData } from "./utils"

const NavBar = ({children}) =>
    <nav>
        <Link to="/"><button>Home</button></Link>
        <Link to="/careers"><button>Careers</button></Link>
        <Link to="/game"><button>Game</button></Link>
        <Link to="/shop"><button>Shop</button></Link>
        {children}
    </nav>

const CartButton = ({ items = 0 }) =>
    <Link to="/cart">
        <button>
            <Icon icon="mdi:cart"></Icon>
            {items}
        </button>
    </Link>

export const App = () => {
    const [cart, setCart] = useState([])
    const [products, setProducts] = useState([])

    const loadData = async () => {
        const data = await getData('https://fakestoreapi.com/products')
        setProducts(data)
    }

    useEffect(() => {
        loadData();
    }, [])

    return (
        <>
            <NavBar>
                <CartButton items={cart.length == 0 ? 0 : cart.map(basket => basket.quantity).reduce((acc, obj) => parseInt(acc) + parseInt(obj))} />
            </NavBar>
            <Outlet context={[cart, setCart, products]}/>
        </>
    )
}