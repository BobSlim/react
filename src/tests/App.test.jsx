import { App } from '@/components/App.jsx'
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';


describe('App', () => {
    it('renders', () => {
        const container = render(<MemoryRouter><App /></MemoryRouter>).container
        expect(container).toMatchSnapshot()
    });
});