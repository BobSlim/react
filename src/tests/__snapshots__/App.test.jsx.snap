// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders 1`] = `
<div>
  <nav>
    <a
      href="/"
    >
      <button>
        Home
      </button>
    </a>
    <a
      href="/careers"
    >
      <button>
        Careers
      </button>
    </a>
    <a
      href="/game"
    >
      <button>
        Game
      </button>
    </a>
    <a
      href="/shop"
    >
      <button>
        Shop
      </button>
    </a>
    <a
      href="/cart"
    >
      <button>
        <span />
        0
      </button>
    </a>
  </nav>
</div>
`;
